scipy/optimize/_zeros_py.py

```diff
@@ -1534,151 +1533,119 @@ def _chandrupatla(f, x0, x1, *, args=(), xrtol=_xtol,
     # which in turn is based on Chandrupatla's algorithm as described in
     # Scherer https://books.google.com/books?id=cC-8BAAAQBAJ&pg=PA95
 
-    # FOR TESTING ONLY
-    if ensure_scalar_x:
-        def f_wrapped(x, *args, **kwargs):
-            return np.asarray([f(x[0], *args, **kwargs)])
-    else:
-        f_wrapped = f
-
-    # FOR TESTING ONLY
-    def termination_function1():
-        """chandrupatla termination per paper"""
-        term = np.logical_or(terminate,
-                             np.logical_or(fm == 0, tlim > 0.5))
-        return term
-
-    def termination_function2():
-        """termination like brent / bisect for testing"""
-        delta = (xatol + xrtol * abs(r))
-        term = np.logical_or(terminate,
-                             np.logical_or(fm == 0,
-                                           abs(b - a) < delta))
-        return term
-
-    termination_function = (termination_function2
-                            if test_switch
-                            else termination_function1)
-
-    # Initialization
-    b = x0
-    a = c = x1
-    fb = f_wrapped(b, *args)
-    fa = fc = f_wrapped(a, *args)
-    t = 0.5
-    iterations = 0
+    def check_convergence(x1, f1, x2, f2, flag, eps_r=1e-5, eps_a=1e-10):
```

Comment:
Add the standard convergence criterion.

Reply:
And then change `test_iterations_expected` to use the callback to terminate according to Chandrupatla's criterion.